### ISSUE TITLE
mix absinthe.schema.json --schema no longer works in 1.4.6

### ISSUE THREAD

**Issue by jhchen** [absinthe-graphql/absinthe#487]
To reproduce:

1. Clone a clean version of https://github.com/absinthe-graphql/absinthe_tutorial
2. Change the Absinthe version in mix.exs to 1.4.6
3. `mix deps.get`
4. `mix absinthe.schema.json --schema BlogWeb.Schema`

Skipping step 2 (and using 1.4.0) results in no error. Specifying 1.4.5 also results in no error so it seems some change in 1.4.6 is the culprit.

### Environment

Elixir 1.5.1
Absinthe 1.4.6

### Expected behavior

schema.json to be generated

### Actual behavior

Mix task fails and no schema.json is generated. Relevant stack trace:

```
$ mix absinthe.schema.json --schema BlogWeb.Schema
==> absinthe
Compiling 2 files (.erl)
Compiling 216 files (.ex)
warning: "not expr1 in expr2" is deprecated. Use "expr1 not in expr2" instead
  lib/absinthe/phase/document/validation/scalar_leafs.ex:69

warning: "not expr1 in expr2" is deprecated. Use "expr1 not in expr2" instead
  lib/absinthe/schema/rule/default_enum_value_present.ex:44

Generated absinthe app
==> absinthe_plug
Compiling 18 files (.ex)
Generated absinthe_plug app
==> absinthe_ecto
Compiling 1 file (.ex)
Generated absinthe_ecto app
** (ArgumentError) argument error
    :erlang.binary_to_existing_atom("Elixir.BlogWeb.Schema", :utf8)
    (elixir) src/elixir_aliases.erl:119: :elixir_aliases.safe_concat/1
    (absinthe) lib/mix/tasks/absinthe.schema.json.ex:55: Mix.Tasks.Absinthe.Schema.Json.run/1
    (mix) lib/mix/task.ex:314: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:80: Mix.CLI.run_task/2
```

**Comment by bruce** [comment]
Reproduced. Regression related to a change introduced by 74d7e9d1985666a1b3cdb169b1bebaef78bae5b2.

**Comment by benwilson512** [comment]
I just looked at what the `ecto.migrate` task does, because it also doesn't require an app start, but it DOES require a working / compiled module in the project (the repo). It has `Mix.Task.run "loadpaths", args` in the task, which we should add too I think.

**Comment by rodrigooler** [comment]
The same problem occurs in the Absinthe v 1.4.7

**Comment by adri** [comment]
Configuring the schema in the `config/config.exs` worked for me.

```elixir
config :absinthe,
  schema: My.Graphql.Schema
```

```bash
mix absinthe.schema.json schema.json
* creating .
* creating schema.json
```

**Comment by adri** [comment]
In `MIX_ENV=test` I got it working using custom alias that calls `loadpaths` in `mix.exs`:

```diff
  defp aliases do
    [
       ....
+      "graphql.schema": ["loadpaths", "absinthe.schema.json"],
       ....
    ] 
  end
```

Example commit: https://github.com/adri/picape/commit/52de502726624148985b84fbace101a746d39ba3

**Comment by silasjmatson** [comment]
I'm running into this same issue, yet when I configure the schema as suggested by @adri, I run into another error:

```
☯ mix absinthe.schema.json
** (UndefinedFunctionError) function MyApp.Schema.context/1 is undefined (module MyApp.Schema is not available)
    MyApp.Schema.context(%{})
    (absinthe) lib/absinthe/phase/schema.ex:39: Absinthe.Phase.Schema.update_context/2
    (absinthe) lib/absinthe/phase/schema.ex:30: Absinthe.Phase.Schema.run/2
    (absinthe) lib/absinthe/pipeline.ex:268: Absinthe.Pipeline.run_phase/3
    (absinthe) lib/absinthe.ex:101: Absinthe.run/3
    (absinthe) lib/mix/tasks/absinthe.schema.json.ex:61: Mix.Tasks.Absinthe.Schema.Json.run/1
    (mix) lib/mix/task.ex:314: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:80: Mix.CLI.run_task/2
```

I'm on absinthe `v1.4.10`

I can run the mix task directly once the app is loaded, in an iex shell or in a test:

```
iex(1)> Mix.Task.run("absinthe.schema.json")
* creating .
* creating ./schema.json
:ok
```

```elixir
defmodule MyApp.GraphQLTest do
  use ExUnit.Case, async: true

  # @moduletag :integration

  setup do
    Application.ensure_all_started(:my_app)
  end

  test "graphql schema passes linting" do
    Mix.Task.run("absinthe.schema.json")
  end
end
```

**Comment by xtian** [comment]
I’m seeing the same behavior as @silasjmatson. The loadpaths alias fix suggested by @adri is working for me, though.

**Comment by ukasiu** [comment]
@silasjmatson I noticed that when environment loads improperly (e.g. not working SQL server) it doesn't work, otherwise it does, at least for me.

**Comment by kreba** [comment]
We also ran into this.

We are now setting the schema in `config.exs` but that alone did not alleviate the issue.
The least invasive (if not very elegant) workaround we found is to run the task as follows:
```
mix do app.start, absinthe.schema.json priv/static/schema.json
```

**Comment by benwilson512** [comment]
Should be fixed in v1.4.12
